### DISCUSSION TITLE
FileUpload not working in Spring Boot

### DISCUSSION THREAD

**Discussion by spierjoe** [primefaces/community#989] [PrimeFaces] [answered]
I working with spring-boot 3.1.4; primefaces 13.0.5-jakarta; jakarta faces 3.0.5; Java SE 17

Listener method is never invoked 

```
<p:fileUpload 
	mode="advanced"
	allowTypes="/(\.|\/)(jpe?g|png)$/" 
	sizeLimit="52428800"
	invalidSizeMessage="Dateigröße darf nicht mehr als 50 MB betragen"
	invalidFileMessage="falscher Dateityp"
	update="messages :druckvorlageuploadform" 
	listener="#{druckFormularAEGUploadBean.handleFileUpload}" />
```
	
or UploadedFile is null 

`<p:fileUpload value="#{fileUploadBean.file}" mode="simple" />`

**Comment by melloware** [comment]
https://stackoverflow.com/questions/68940746/unstable-primefaces-fileupload-listener-call  Make sure you have removed Apache Commons FileUpload from the WAR and from `web.xml` you should be using `native` uploading which uses Jakarta Servlet native uploading.

**Reply by spierjoe** [reply]
unfortunately it didn't help.

I tried in pom.xml it with:
```
<dependency>
    <groupId>commons-fileupload</groupId>
    <artifactId>commons-fileupload</artifactId>
    <version>1.5</version>
</dependency>
```

or with:
```
<dependency>
    <groupId>org.apache.commons</groupId>
    <artifactId>commons-fileupload2-jakarta</artifactId>
    <version>2.0.0-M1</version>
</dependency>

```
or without upload-jar.

**Reply by melloware** [reply]
You shouldn't need any of that it should work with Servlet native mode and not Apache Commons Upload. 

However on StackOverflow you will find tons of posts like this: https://stackoverflow.com/questions/27552227/primefaces-fileupload-not-working-in-spring-boot

its a Spring Boot filter issue you need to figure out what in Spring Boot is blocking the upload.

**Comment by spierjoe** [comment, marked as answer]
Unfortunately, some of the references were a little confusing.
This reference helped best:
https://forum.primefaces.org/viewtopic.php?t=57347
Although the entries are for the web.xml, they only need to be applied to springboot.

```
    @Override
    public void setServletContext(ServletContext servletContext) {
        servletContext.setInitParameter("com.sun.faces.forceLoadConfiguration", Boolean.TRUE.toString());
        servletContext.setInitParameter("jakarta.faces.FACELETS_SKIP_COMMENTS", "true");
        servletContext.setInitParameter("primefaces.UPLOADER", "native");
    }
```
